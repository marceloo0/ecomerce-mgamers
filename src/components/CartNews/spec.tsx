import { screen } from '@testing-library/react'
import { renderWithTheme } from '../../utils/tests/helpers'

import { CartNews } from '.'

const props = {
  item: {
    id: 1,
    img: '/img/products/mouseRazer.svg',
    price: 180.94,
    promotion: 155.9,
    title: 'Mouse Gamer Razer Deathadder',
    titleBackground: 'MouseS',
    quantity: 0
  }
}

describe('<CartNews />', () => {
  it('should render default', () => {
    const { container } = renderWithTheme(<CartNews {...props} />)

    expect(
      screen.getByText(/Mouse Gamer Razer Deathadder/i)
    ).toBeInTheDocument()

    expect(screen.getByRole('img', { name: props.item.img })).toHaveAttribute(
      'src',
      props.item.img
    )

    expect(container.firstChild).toMatchSnapshot()
  })
})
