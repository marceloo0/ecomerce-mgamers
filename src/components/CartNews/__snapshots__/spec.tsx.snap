// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartNews /> should render default 1`] = `
.c7 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-family: Poppins,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  border: 1px solid transparent;
  border-radius: 0.4rem;
  cursor: pointer;
  color: #C7ECEC;
  background-color: #201C2B;
  padding: 0.5rem;
}

.c7:hover {
  background-color: #332756;
}

.c7 > svg {
  color: #7FFA45;
  margin-right: 0.5rem;
}

.c0 {
  padding: 24px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
}

.c1 {
  width: 100%;
  height: 290px;
  left: auto;
  top: 0;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 36px;
}

.c6 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
}

.c3 {
  color: #C7ECEC;
  width: 200px;
  text-align: center;
  line-height: 18px;
}

.c4 {
  color: #415151;
  font-size: 1.6rem;
  -webkit-text-decoration-line: line-through;
  text-decoration-line: line-through;
  line-height: 15px;
}

.c5 {
  color: #25AF43;
  font-size: 1.8rem;
  line-height: 13px;
}

<div
  class="c0"
>
  <img
    alt="/img/products/mouseRazer.svg"
    class="c1"
    src="/img/products/mouseRazer.svg"
  />
  <div
    class="c2"
  >
    <h3
      class="c3"
    >
      Mouse Gamer Razer Deathadder
    </h3>
    <span
      class="c4"
    >
      180.94
    </span>
    <h3
      class="c5"
    >
      155.9
    </h3>
  </div>
  <div
    class="c6"
  >
    <button
      class="c7"
    >
      <svg
        fill="currentColor"
        height="24"
        stroke="currentColor"
        stroke-width="0"
        version="1.1"
        viewBox="0 0 16 16"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M15.5 6h-5.5v-5.5c0-0.276-0.224-0.5-0.5-0.5h-3c-0.276 0-0.5 0.224-0.5 0.5v5.5h-5.5c-0.276 0-0.5 0.224-0.5 0.5v3c0 0.276 0.224 0.5 0.5 0.5h5.5v5.5c0 0.276 0.224 0.5 0.5 0.5h3c0.276 0 0.5-0.224 0.5-0.5v-5.5h5.5c0.276 0 0.5-0.224 0.5-0.5v-3c0-0.276-0.224-0.5-0.5-0.5z"
        />
      </svg>
      <span>
        Adicionar
      </span>
    </button>
  </div>
</div>
`;
